### PR TITLE
QOLDEV-819 add autoscaling groups alongside fixed instances

### PR DESCRIPTION
**Pull request by ThrawnCA**
Once the autoscaling instances are live, we can decommission the OpsWorks instances.

### CODE REVIEW

File: templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2 [hunk 1]
```diff
@@ -501,4 +507,94 @@ Resources:
...
+              VolumeType: "gp2"
+          - DeviceName: "/dev/sdi"
+            Ebs:
+              DeleteOnTermination: true
+              VolumeSize: 32
+              VolumeType: "gp2"
+        IamInstanceProfile:
+          Arn:
+            Fn::ImportValue: !Sub "${Environment}${ApplicationName}WebInstanceProfile"
+        ImageId: ami-03b836d87d294e89e
```
Comment (duttonw) [inline templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2:534]: What image is this?

File: templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2 [hunk 2]
```diff
@@ -501,4 +507,94 @@ Resources:
...
+          - Fn::ImportValue: !Sub "${Environment}CKANManagementSG"
+          - Fn::ImportValue: !Sub "${Environment}CKANAppAsgSG"
+        UserData:
+          Fn::Base64:
+            Fn::Sub: |
+              #!/bin/sh
+              if ! (grep '/mnt/local_data' /etc/fstab >/dev/null); then
+                mkdir /mnt/local_data
+                mkfs -t xfs /dev/sdi
+                echo '/dev/sdi /mnt/local_data xfs defaults,nofail 0 2' >> /etc/fstab
```
Comment (duttonw) [inline templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2:547]: can't we move back to only using the primary disk at 100gb now?

File: templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2 [hunk 3]
```diff
@@ -501,4 +507,94 @@ Resources:
...
+                mount -a
+              fi
+              metadata_token=`curl -X PUT -H "X-aws-ec2-metadata-token-ttl-seconds: 60" http://169.254.169.254/latest/api/token` && \
+              INSTANCE_ID=$(curl -H "X-aws-ec2-metadata-token: $metadata_token" http://169.254.169.254/latest/meta-data/instance-id) && \
```
Comment (duttonw) [inline templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2:551]: 👍

File: templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2 (hunk 2 above)
Comment (ThrawnCA) [reply templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2:547]: Having a separate disk gives us an extra layer of protection; filling the secondary disk doesn't crash the whole server in the way that filling the root disk would do.

**Review by duttonw — APPROVED** [review]

File: templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2 (hunk 1 above)
Comment (ThrawnCA) [reply templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2:534]: It should probably be a parameter, I suppose.

It's a public AMI of Amazon Linux 2 (ie what OpsWorks has been using).

File: templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2 (hunk 2 above)
Comment (duttonw) [reply templates/Datashades-OpsWorks-CKAN-Instances.cfn.yml.j2:547]: does mean needing to monitor both. also with the move to ASG, the instances will be rotated more often than disk getting filled.

**Review by duttonw — APPROVED** [review]

**Pull request merged**